activemodel/lib/active_model/api.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  # == Active \Model \API
+  #
+  # Includes the required interface for an object to interact with
+  # Action Pack and Action View, using different Active Model modules.
+  # It includes model name introspections, conversions, translations and
+  # validations. Besides that, it allows you to initialize the object with a
+  # hash of attributes, pretty much like Active Record does.
+  #
+  # A minimal implementation could be:
+  #
+  #   class Person
+  #     include ActiveModel::API
+  #     attr_accessor :name, :age
+  #   end
+  #
+  #   person = Person.new(name: 'bob', age: '18')
+  #   person.name # => "bob"
+  #   person.age  # => "18"
+  #
+  # Note that, by default, <tt>ActiveModel::API</tt> implements <tt>persisted?</tt>
+  # to return +false+, which is the most common case. You may want to override
+  # it in your class to simulate a different scenario:
+  #
+  #   class Person
+  #     include ActiveModel::API
+  #     attr_accessor :id, :name
+  #
+  #     def persisted?
+  #       self.id == 1
```

Comment:
From the perspective of someone newer to programming/Rails, this may be a little confusing. What are your thoughts about adjusting this documentation to be less specific to the instance created below?

In the documented example, `person.persisted?` would only return `true` if `id == 1`. I would assume here that should someone be implementing this in their application - that they'd be checking against the existence of the `id` instead of its value.

```ruby
def persisted?
  self.id.present?
end
```

```suggestion
  #       self.id.present?
```

---

As a side - It looks like the documentation was copied over from `ActiveModel::Model`, and given the purpose of the PR, I wouldn't expect the entirety of each piece of documentation to be reviewed.

I just happened to be looking for something else and saw this pop-up. No stress if you feel that my suggestion is out of scope.

Reply:
@thewatts No that is a good call, I was just pairing with @p8 on this and we're figuring out the scope of this PR and how we can split up the work to build an AM::API with attributes. This API is currently private so before we make it public we have to put enough care into the design. Appreciate any ideas or feedback along the way! :bow:

Reply:
I agree 😄

Reply:
@thewatts I’ve made the change and added you as a co author.